File: PokerAppBackend/Contracts/PlayerDto.cs
namespace PokerAppBackend.Contracts;

public sealed class PlayerDto
{
    public int SeatIndex { get; init; }
    public string Name { get; init; } = "";
    public bool IsBot { get; init; }
    public bool HasFolded { get; init; }
    public List<CardDto> Hole { get; init; } = new();
    public List<string> LegalActions { get; init; } = new();
}

File: react-frontend/src/api/dto/PlayerDto.ts
import { CardDto } from "./CardDto";
import { PlayerAdvisoryDto } from "./PlayerAdvisoryDto";

export interface PlayerDto {
    seatIndex: number;
    name: string;
    isBot: boolean;
    hasFolded: boolean;
    hole: CardDto[];
    legalActions: string[];
    lastestAction: string;
    isOut: boolean;
    stack: number;
    committedThisStreet: number;
    committedThisHand: number;
    playerAdvisory?: PlayerAdvisoryDto | null;
}
